libcontainer/sync_unix.go

```diff
@@ -14,7 +14,8 @@ import (
 // which ends up making things like json.Decoder hang forever if the packet is
 // bigger than the internal read buffer.
 type syncSocket struct {
-	f *os.File
+	f         *os.File
+	procReady bool
```

Comment:
I don't like that this is put inside `syncSocket` because:

1. `procReady` only makes sense for one of our pipes, but we have multiple.
2. With this patch, `procReady` means two different things for `runc init` and the parent runc process. Combining the logic makes it far more confusing.

I think having a `closed bool` that is only be used for `runc init` might make some sense, though I don't like that access to the bool isn't protected by a mutex (then again, I suspect we wouldn't concurrently close it so :shrug:.)